pkgs/games/hmcl/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib, stdenv
+, fetchurl
+, jre
+, makeWrapper
+, makeDesktopItem
+, copyDesktopItems
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hmcl";
+  version = "3.5.3.223";
+
+  src = ./.;
+
+  hmcl = fetchurl {
+    url = "https://github.com/huanghongxun/HMCL/releases/download/v${version}/HMCL-${version}.jar";
+    sha256 = "sha256-8g2FMvAiAKfxJUY0G7wl6d44wOpVsknFHGZ85IkOzFc=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [ makeWrapper copyDesktopItems ];
+
+  desktopItem = makeDesktopItem {
+    name = "hmcl";
+    exec = "hmcl";
+    icon = "hmcl";
+    comment = "A Minecraft Launcher which is multi-functional, cross-platform and popular";
+    desktopName = "HMCL";
+    categories = [ "Game" ];
+  };
+
+  desktopItems = [ desktopItem ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/opt/hmcl
+    install -Dm644 $hmcl $out/opt/hmcl/hmcl.jar
+    install -Dm644 ${./icon.png} $out/share/icons/hicolor/48x48/apps/hmcl.png
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    makeWrapper ${jre}/bin/java $out/bin/hmcl \
+      --argv0 "hmcl" \
+      --add-flags "-jar $out/opt/hmcl/hmcl.jar" \
+      --prefix PATH : ${lib.makeBinPath [ jre ]} \
```

Comment:
Is `jre` really needed in `PATH`? Does the program recursively call itself or something?

Reply:
This is what allows ```hmcl``` to find `jre` and launch Minecraft

Reply:
> Is `jre` really needed in `PATH`? Does the program recursively call itself or something?

This is what allows `hmcl` to find `jre` and launch Minecraft